packages/ifpack2/src/Ifpack2_BlockTriDiContainer_decl.hpp

```diff
@@ -54,315 +54,339 @@
+      //! factorization, such that the entry is moved radially outward in the
+      //! complex plane. N.B. that this constant modifies the matrix in the linear
+      //! equation, not simply the diagonal preconditioner.
+      magnitude_type addRadiallyToDiagonal;
+      ComputeParameters();
+    };
+
+    /// \brief Create a ComputeParameters struct with default values.
+    ComputeParameters createDefaultComputeParameters () const;
+
+    /// \brief Extract the local tridiagonal block and prepare the solver.
+    ///
+    /// This version of <tt>applyInverseJacobi</tt> is meant to be called by
+    /// direct users of this class, rather than by <tt>BlockRelaxation</tt>.
+    ///
+    /// \param addRadiallyToDiagonal [in] Add a constant to each diagonal entry of
+    ///   the matrix. It is added according to the (possibly complex) sign of the
+    ///   digonal entry at the time that the entry is the pivot in the LU
+    ///   factorization, such that the entry is moved radially outward in the
+    ///   complex plane. N.B. that this constant modifies the matrix in the linear
+    ///   equation, not simply the diagonal preconditioner.
+    void compute (const ComputeParameters& input);
+
+    //! Input arguments to <tt>applyInverseJacobi</tt>
+    struct ApplyParameters {
+      //! Set this to true if <tt>Y</tt> is meant to be treated as 0 on
+      //! entry. Defaults to false.
+      bool zeroStartingSolution;
+      //! Damping factor. Defaults to 1.
+      scalar_type dampingFactor;
+      //! The maximum number of sweeps. If the norm-based criterion is not used,
+      //! it's exactly the number of sweeps. Defaults to 1.
+      int maxNumSweeps;
+      //! The 2-norm-based termination criterion. If it is larger than 0, then it
+      //! is used. Termination is then based on <tt>maxNumSweeps</tt> and this
+      //! criterion, whichever is reached first. In the case of multiple right
+      //! hand sides, norm-based termination occurs only when each vector
+      //! satisfies the criterion. The criterion for a vector is <tt>f(D^{-1} (x -
+      //! A*y)) <= tolerance * f(D^{-1} (x - R*y_0))</tt>, where <tt>y_0</tt> is
+      //! the input <tt>y</tt>, often 0, and <tt>f</tt> is the maximum of the
+      //! 2-norms of each degree of freedom, i.e., index of a block. Defaults to
+      //! 0.
+      magnitude_type tolerance;
+      //! Check the norm-based termination criterion every
+      //! <tt>checkToleranceEvery</tt> iterations. Defaults to 1. A norm
+      //! computation requires a global reduction, which is expensive. Hence it
+      //! can make sense to check for termination only every 2 or 3 iterations if
+      //! it is anticipated that the number of iterations required to terminate
+      //! will be sufficiently large to make 1 or 2 extra iterations relatively
+      //! cheap. Defaults to 1.
+      int checkToleranceEvery;
+      ApplyParameters();
```

Comment:
You could just set the default values directly; you don't need to define a separate default constructor.  For example:
```
struct Foo {
  int bar = 42;
};
```